src/DotLiquid/StandardFilters.cs

```diff
@@ -778,12 +792,19 @@ public static object Minus(Context context, object input, object operand)
         /// <param name="operand">Number to multiple input by</param>
         public static object Times(Context context, object input, object operand)
         {
-            if (context.SyntaxCompatibilityLevel >= SyntaxCompatibility.DotLiquid21)
-                return DoMathsOperation(context, input, operand, Expression.MultiplyChecked);
+            try
+            {
+                if (context.SyntaxCompatibilityLevel >= SyntaxCompatibility.DotLiquid21)
+                    return DoMathsOperation(context, input, operand, Expression.MultiplyChecked);
 
-            return input is string && (operand is int || operand is long)
-                ? Enumerable.Repeat((string)input, Convert.ToInt32(operand))
-                : DoMathsOperation(context, input, operand, Expression.MultiplyChecked);
+                return input is string && (operand is int || operand is long)
+                    ? Enumerable.Repeat((string)input, Convert.ToInt32(operand))
+                    : DoMathsOperation(context, input, operand, Expression.MultiplyChecked);
+            }
+            catch (OverflowException) when (input is int)
+            {
+                return DoMathsOperation(context, Convert.ToInt64(input), operand, Expression.MultiplyChecked);
+            }
         }

```

Comment:
what about divide by fraction? e.g. 2147483647 / 0.5 might result in an overflow.

Reply:
Should convert to a floating point hence range is different.